README.md

```diff
@@ -0,0 +1,117 @@
+but you should go ahead and create keys in advance of the new policy coming
+into effect. Policy should start to be the only way sometime in April 2021.
+
+We use SSH as primary access to all our Datalab infrastructure and as well as
+parts of the OpenSAFELY backend infrastructure. We also use it for Github
+authentication for cloning and pushing. 
+
+Thus, keeping our SSH keys and usage secure is very important. This document
+describes our policy for SSH key management.
+
+
+## The Key
+
+You should have a dedicated Datalab SSH key for use with Datalab systems, and
+not with other systems. This should be a modern ed25519 key, protected by
+a strong passphrase, and have your Datalab email address and creation date in
+the key's comment.
+
+To generate such a key, you can run this in a terminal:
+
+    ssh-keygen -t ed25519 -C "YOUREMAIL@thedatalab.org:$(date --iso-8601)" ~/.ssh/datalab_ed25519
+
+Note: you MUST enter a strong password when prompted.
+
+
+## Add to your Github Account
+
+We use Github as a key distribution mechanism. You should add the contents of
+your public key file (e.g. `~/.ssh/datalab_ed25519.pub`) to your Github
+account:
+
+[https://github.com/settings/keys](https://github.com/settings/keys)
+
+Github doesn't preserve the key comment, so we recommend titling the key with
+the comment, e.g.
+
+     YOURRMAIL@thedatalab.org:202X-XX-XX
+
+
+## Confirm your Key
+
+Clone this repo and run the following to a ensure your new keys fingerprint is
+added:
+
+    ./add.py YOUR_GITHUB_USERNAME ~/.ssh/datalab_ed25519
+
+This is should update your user and key fingerprint in the `passwd` file,
+marking this key as your current Datalab key.
+
+You need to add and commit this, but you MUST sign it with your registered
+Github GPG key, or else the commit will be blocked.
+
+Once this is reviewed and merged, after a short time your account and key
+should be available to log in with on all our servers.
+
+
+## Local SSH Config
+
+You can use your local SSH config to ensure you use the right username and key when
+accessing Datalab systems with the following snippet in `~/.ssh/config`
+
+
+```
+Host *.ebmdatalab.net *.opensafely.org
+    User YOUR_GITHUB_USERNAME
+    IdentityFile ~/.ssh/datalab_ed25519
+```
+
+This should allow you to just do `ssh somehost.embdatalab.net` and it Just Works.
+
+
+## Key Compromise and Rotation
+
+If your key is ever compromised (e.g. your computer is stolen or hacked), you
+should inform Simon or Tom W ASAP, and immediately remove the public key from
+your Github account, and also this repo. Either of these actions will cause the
+key to be removed from our various systems within a short window.
+
+You can always add a new key at any time by generating it as above, and adding
+it to Github and the fingerprint to this repo as described. You can then remove
+the old key, and it will be removed from our systems.  You are encouraged to
```

Comment:
from my reading of the script, when you add the fingerprint to this repo, the old fingerprint will be removed? if so we should probably tweak this to avoid mild confusion.

Reply:
will do